connectors/source.py

```diff
@@ -113,7 +113,15 @@ def __init__(self, configuration):
         self.configuration.set_defaults(self.get_default_configuration())
 
     def __str__(self):
-        return f"Datasource `{self.__class__.__doc__}`"
+        return f"Datasource `{self.__class__.name()}`"
+
+    @staticmethod
```

Comment:
```suggestion
    @classmethod
```

Should this be `@classmethod`(s)? I think `@staticmethod`(s) are similar to top-level functions in a python module.